pages/api/guestbook/[id].js

```diff
@@ -1,25 +1,31 @@
-import redis from '@/lib/redis';
+import db from '@/lib/planetscale';
 import session from '@/lib/session';
 
 export default async function handler(req, res) {
   session(req, res);
 
   const { id } = req.query;
-  const { login, email } = req.session;
-  const entry = JSON.parse((await redis.hget('guestbook', id)) || 'null');
+  const { login } = req.session;
 
-  if (req.method === 'GET') {
-    const { email, ...restOfEntry } = entry;
+  const [rows] = await db.query(`
+    SELECT id, body, created_by FROM guestbook
+    WHERE id = "${id}";
+  `);
```

Comment:
There are a few places here that use string concatenation with user-controlled values to build a SQL query, which leads to SQL injection vulnerabilities. Passing an array of parameter values allows the mysql2 driver to sanitize the values before querying the database.

```suggestion
  const [rows] = await db.query(`
    SELECT id, body, created_by FROM guestbook
    WHERE id = ?;
  `, [id]);
```

Reply:
Good call! Would love to eventually switch this to Prisma on top, but this is a great improvement 😄